### PR TITLE
Update doxygen documentation

### PR DESCRIPTION
**Pull request by bradleysmith23**
*Issue #, if available:*

*Description of changes:*
Update doxygen documentation to reflect the addition of API's to jobs.c and the addition of the Ota Job Parser

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by manvensh — APPROVED** [review]

File: source/include/jobs.h [hunk 1]
```diff
@@ -633,11 +644,39 @@ JobsStatus_t Jobs_StartNext( char * buffer,
...
+ *     // MQTT client of your choice.
+ * }
+ * @endcode
  */
+/* @[declare_jobs_startnextmsg] */
 size_t Jobs_StartNextMsg( const char * clientToken,
                           size_t clientTokenLength,
                           char * buffer,
                           size_t bufferSize );
+/* @[declare_jobs_startnextmsg] */
```
Comment (Skptak) [inline source/include/jobs.h:679]: Is this a duplicate line or intended?

File: source/otaJobParser/include/job_parser.h [hunk 2]
```diff
@@ -14,21 +14,25 @@
...
+    size_t filepathLen;      /**< @brief Length of filepath */
+    const char * certfile;   /**< @brief Path to Code Signing Certificate on Device */
+    size_t certfileLen;      /**< @brief Length of certfile */
+    const char * authScheme; /**< @brief Authentication Scheme for HTTP URL ( null for MQTT  )*/
+    size_t authSchemeLen;    /**< @brief Length of authScheme */
+    const char * imageRef;   /**< @brief MQTT Stream or HTTP URL */
+    size_t imageRefLen;      /**< @brief Length of imageRef */
+    uint32_t fileId;         /**< @brief File ID */
+    uint32_t fileSize;       /**< @brief Size of the OTA Update */
+    uint32_t fileType;       /**< @brief FIle Type */
```
Comment (Skptak) [inline source/otaJobParser/include/job_parser.h:35]: I personally really dislike same line comments like this. They tend to line wrap very easily, which then just ends up being a worse version of having the comments on their own line anyways.
I'd much rather these comments be on the line before.

File: source/otaJobParser/include/ota_job_processor.h [hunk 3]
```diff
@@ -17,9 +17,42 @@
 
 #include "job_parser.h"
 
+/**
+ * @brief Signals if the job document provided is a FreeRTOS OTA update document
```
Comment (Skptak) [inline source/otaJobParser/include/ota_job_processor.h:21]:
```suggestion
 * @brief Signals if the job document provided is an AWS IoT Core OTA update document
```

File: source/otaJobParser/include/job_parser.h (hunk 2 above)
Comment (Skptak) [reply source/otaJobParser/include/job_parser.h:35]:
```suggestion    /* @brief Code Signing Signature */
    const char * signature;

    /* @brief Length of signature */
    size_t signatureLen;

    /* @brief File path to store OTA Update on device */
    const char * filepath;

    /* @brief Length of filepath */
    size_t filepathLen;

    /* @brief Path to Code Signing Certificate on Device */
    const char * certfile;

    /* @brief Length of certfile */
    size_t certfileLen;

    /* @brief Authentication Scheme for HTTP URL ( null for MQTT ) */
    const char * authScheme;

    /* @brief Length of authScheme */
    size_t authSchemeLen;

    /* @brief MQTT Stream or HTTP URL */
    const char * imageRef;

    /* @brief Length of imageRef */
    size_t imageRefLen;

    /* @brief File ID */
    uint32_t fileId;

    /* @brief Size of the OTA Update */
    uint32_t fileSize;

    /* @brief FIle Type */
    uint32_t fileType;

```

File: source/otaJobParser/include/job_parser.h [hunk 4]
```diff
@@ -14,21 +14,25 @@
...
-    uint32_t fileId;
-    uint32_t fileSize;
-    uint32_t fileType;
+    const char * signature;  /**< @brief Code Signing Signature */
+    size_t signatureLen;     /**< @brief Length of signature */
+    const char * filepath;   /**< @brief File path to store OTA Update on device */
+    size_t filepathLen;      /**< @brief Length of filepath */
+    const char * certfile;   /**< @brief Path to Code Signing Certificate on Device */
+    size_t certfileLen;      /**< @brief Length of certfile */
+    const char * authScheme; /**< @brief Authentication Scheme for HTTP URL ( null for MQTT  )*/
```
Comment (Skptak) [inline source/otaJobParser/include/job_parser.h:29]:
```suggestion
    const char * authScheme; /**< @brief Authentication Scheme for HTTP URL ( null for MQTT ) */
```

**Review by manvensh — APPROVED** [review]

**Review by Skptak — APPROVED** [review]

**Pull request merged**